### DISCUSSION TITLE
SWR support header parameters?

### DISCUSSION THREAD

**Discussion by rootlinux2** [vercel/swr#466] [General] [answered]
How can I pass header parameters like auth_token?

**Comment by sergiodxa** [comment, marked as answer]
In your fetcher you can use any header you want, if you want to receive them dynamically you can pass an array as key of useSWR

```js
function fetcher(url, token) {
  // fetch
}

useSWR([endpoint, token], fetcher)
```

**Reply by rootlinux2** [reply]
Thank you!
